rocksdb.nim

```diff
@@ -1,236 +1,12 @@
-    res.ok(string.fromBytes(data))
-
-  if ? db.get(key, onData):
-    res
-  else:
-    ok("")
-
-proc getBytes*(db: RocksDBInstance, key: openArray[byte]): RocksDBResult[seq[byte]]  {.deprecated: "DataProc".} =
-  ## Get value for `key`. If no value exists, set `result.ok` to `false`,
-  ## and result.error to `""`.
-  var res: RocksDBResult[seq[byte]]
-  proc onData(data: openArray[byte]) =
-    res.ok(@data)
-
-  if ? db.get(key, onData):
-    res
-  else:
-    err("")
-
-proc put*(db: RocksDBInstance, key, val: openArray[byte]): RocksDBResult[void] =
-  if key.len <= 0:
-    return err("rocksdb: key cannot be empty on put")
-
-  var
-    errors: cstring
-
-  rocksdb_put(db.db, db.writeOptions,
-              cast[cstring](unsafeAddr key[0]), csize_t(key.len),
-              cast[cstring](if val.len > 0: unsafeAddr val[0] else: nil),
-              csize_t(val.len),
-              errors.addr)
-
-  bailOnErrors()
-  ok()
-
-proc contains*(db: RocksDBInstance, key: openArray[byte]): RocksDBResult[bool] =
-  if key.len <= 0:
-    return err("rocksdb: key cannot be empty on contains")
-
-  var
-    errors: cstring
-    len: csize_t
-    data = rocksdb_get(db.db, db.readOptions,
-                       cast[cstring](unsafeAddr key[0]), csize_t(key.len),
-                       addr len, errors.addr)
-  bailOnErrors()
-  if not data.isNil:
-    rocksdb_free(data)
-    ok(true)
-  else:
-    ok(false)
-
-proc del*(db: RocksDBInstance, key: openArray[byte]): RocksDBResult[bool] =
-  if key.len <= 0:
-    return err("rocksdb: key cannot be empty on del")
-
-  # This seems like a bad idea, but right now I don't want to
-  # get sidetracked by this. --Adam
-  if not db.contains(key).get:
-    return ok(false)
-
-  var errors: cstring
-  rocksdb_delete(db.db, db.writeOptions,
-                  cast[cstring](unsafeAddr key[0]), csize_t(key.len),
-                  errors.addr)
-  bailOnErrors()
-  ok(true)
-
-proc clear*(db: var RocksDBInstance): RocksDBResult[bool] =
-  raiseAssert "unimplemented"
-
-proc backup*(db: RocksDBInstance): RocksDBResult[void] =
-  var errors: cstring
-  rocksdb_backup_engine_create_new_backup(db.backupEngine, db.db, errors.addr)
-  bailOnErrors()
-  ok()
-
-# XXX: destructors are just too buggy at the moment:
-# https://github.com/nim-lang/Nim/issues/8112
-# proc `=destroy`*(db: var RocksDBInstance) =
-proc close*(db: var RocksDBInstance) =
-  template freeField(name) =  
-    type FieldType = typeof db.`name`
-    if db.`name`.isNil:
-      `rocksdb name destroy`(db.`name`)
-      db.`name` = FieldType(nil)
-  template setFieldToNil(name) =
-    type FieldType = typeof db.`name`
-    db.`name` = FieldType(nil)
-  
-  freeField(writeOptions)
-  freeField(readOptions)
-  freeField(options)
-
-  if not db.backupEngine.isNil:
-    rocksdb_backup_engine_close(db.backupEngine)
-    setFieldToNil(backupEngine)
-
-  if not db.db.isNil:
-    rocksdb_close(db.db)
-    setFieldToNil(db)
+export backup, rocksdb
```

Comment:
Add a newline to all files where one is missing (perhaps set-up your editor for this).

Reply:
Sure, will do. I thought I had already configured this in vscode but apparently I missed that setting. Done now.